src/Cdn77/Sniffs/NamingConventions/ValidVariableNameSniff.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Cdn77\Sniffs\NamingConventions;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\AbstractVariableSniff;
+
+use function assert;
+use function ltrim;
+use function preg_match;
+use function preg_match_all;
+use function sprintf;
+
+use const T_DOUBLE_COLON;
+use const T_NULLSAFE_OBJECT_OPERATOR;
+use const T_OBJECT_OPERATOR;
+use const T_OPEN_PARENTHESIS;
+use const T_STRING;
+use const T_WHITESPACE;
+
+class ValidVariableNameSniff extends AbstractVariableSniff
+{
+    public const CODE_DOES_NOT_MATCH_PATTERN = 'DoesNotMatchPattern';
+    public const CODE_MEMBER_DOES_NOT_MATCH_PATTERN = 'MemberDoesNotMatchPattern';
+    public const CODE_STRING_DOES_NOT_MATCH_PATTERN = 'StringDoesNotMatchPattern';
+    private const PATTERN_CAMEL_CASE_OR_UNUSED = '\b(([a-zA-Z][a-zA-Z0-9]*?([A-Z][a-zA-Z0-9]*?)*?)|_+)\b';
+
+    public string $pattern = self::PATTERN_CAMEL_CASE_OR_UNUSED;
+    public string $memberPattern = self::PATTERN_CAMEL_CASE_OR_UNUSED;
+    public string $stringPattern = self::PATTERN_CAMEL_CASE_OR_UNUSED;
```

Comment:
member and string vars imho shouldnt be unused (`$_`).

Reply:
Good point [81b865c](https://github.com/cdn77/coding-standard/pull/66/commits/81b865cfa57dc1b57f53093c4e71280bed03d5d6)